atom/browser/browser_mac.mm

```diff
@@ -217,14 +217,76 @@
   return settings;
 }
 
+// copied from GetLoginItemForApp in base/mac/mac_util.mm
+LSSharedFileListItemRef GetLoginItemForApp() {
+  base::ScopedCFTypeRef<LSSharedFileListRef> login_items(
+      LSSharedFileListCreate(NULL, kLSSharedFileListSessionLoginItems, NULL));
+  if (!login_items.get()) {
+    LOG(ERROR) << "Couldn't get a Login Items list.";
+    return NULL;
+  }
+  base::scoped_nsobject<NSArray> login_items_array(
+      base::mac::CFToNSCast(LSSharedFileListCopySnapshot(login_items, NULL)));
+  NSURL* url = [NSURL fileURLWithPath:[base::mac::MainBundle() bundlePath]];
+  for (NSUInteger i = 0; i < [login_items_array count]; ++i) {
+    LSSharedFileListItemRef item =
+        reinterpret_cast<LSSharedFileListItemRef>(login_items_array[i]);
+    CFURLRef item_url_ref = NULL;
+    if (LSSharedFileListItemResolve(item, 0, &item_url_ref, NULL) == noErr &&
+        item_url_ref) {
+      base::ScopedCFTypeRef<CFURLRef> item_url(item_url_ref);
+      if (CFEqual(item_url, url)) {
+        CFRetain(item);
+        return item;
+      }
+    }
+  }
+  return NULL;
+}
+
+void RemoveFromLoginItems() {
+  base::ScopedCFTypeRef<LSSharedFileListRef> list(
+      LSSharedFileListCreate(NULL, kLSSharedFileListSessionLoginItems, NULL));
+  if (list) {
+    if (GetLoginItemForApp() != NULL) {
+      CFURLRef url = (__bridge CFURLRef)
+          [NSURL fileURLWithPath:[[NSBundle mainBundle] bundlePath]];
+      if (url) {
+        UInt32 seed;
+        base::ScopedCFTypeRef<CFArrayRef> items(
+            (CFArrayRef)LSSharedFileListCopySnapshot(list, &seed));
+        if (items) {
+          for (id item in (__bridge NSArray*)items.get()) {
+            LSSharedFileListItemRef itemRef =
+                (__bridge LSSharedFileListItemRef)item;
+            if (LSSharedFileListItemResolve(itemRef, 0, &url, NULL) == noErr &&
```

Comment:
this looks like it overwrites `url`, so the initial value that is set on L252 never gets used. So the initial value of `url` can probably be `NULL`, and even better, that variable's declaration can be moved up to the line before this call to `LSSharedFileListItemResolve`

Reply:
Also, `LSSharedFileListItemResolve` looks like it was [deprecated in 10.10](https://developer.apple.com/documentation/coreservices/1447347-lssharedfilelistitemresolve?language=objc), to be replaced by `LSSharedFileListItemCopyResolvedURL` which was then immediately [deprecated in 10.11](https://developer.apple.com/documentation/coreservices/1449882-lssharedfilelistitemcopyresolved?language=objc) without any apparent replacement 😕

Chromium still uses `LSSharedFileListItemResolve` though so I think we're ok for now.